docs/secret-engines.md

```diff
@@ -194,6 +197,130 @@ to read a new credential from this role, execute the following:
 vault read -tls-skip-verify github/raf-backstage-demo/token/one-repo-only
 ```
 
+## QuaySecretEngineConfig
+
+The `QuayHubSecretEngineConfig` CRD allows a user to create a Quay Secret engine configuration. Only one configuration can exists per Quay secret engine mount point, here is an example:
+
+```yaml
+apiVersion: redhatcop.redhat.io/v1alpha1
+kind: QuayHubSecretEngineConfig
+metadata:
+  name: quay-org
+spec:
+  authentication: 
+    path: kubernetes
+    role: policy-admin
+  rootCredentials:
+    secret:
+      name: quay-token
+  path: quay/demo
+  url: https://quay.io
+```
+
+The `path` field specifies the path of the secret engine that will contain this configuration.
+
+The `url` field specifies the endpoint for the Quay server
+
+The token can retrieved in three different ways:
+
+1. From a Kubernetes secret, specifying the `rootCredentialsFromSecret` field. The secret must be of [basic auth type](https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret). If the secret is updated this connection will also be updated.
+2. From a Vault secret, specifying the `rootCredentialsFromVaultSecret` field.
+3. From a [RandomSecret](#RandomSecret), specifying the `rootCredentialsFromRandomSecret` field. When the RandomSecret generates a new secret, this connection will also be updated.
+
+More parameters exists for their explanation and for how to install the vault-plugin-secrets-quay engine see [here](https://github.com/redhat-cop/vault-plugin-secrets-quay)
+
+This CR is roughly equivalent to this Vault CLI command:
+
+```shell
+vault write [namespace/]quay/demo/config url=https://<QUAY_HOST> token=<token>
+```
+
+## QuayHubSecretEngineRole
+
+The `QuayHubSecretEngineRole` CRD allows a user to create a Quay secret engine role. A role allows for the creation of a narrowly scoped Quay Robot account within an organization.
```

Comment:
the `Hub` necessary on the CRD name? It does not match well with the engine config.

Reply:
Bad docs. I updated the content to correct.